DQM/SiPixelPhase1Track/plugins/SiPixelPhase1EfficiencyExtras.cc

```diff
@@ -0,0 +1,223 @@
+
+//------------------------------------------------------------------
+// Method called for every event
+//------------------------------------------------------------------
+void SiPixelPhase1EfficiencyExtras::dqmEndJob(DQMStore::IBooker& iBooker, DQMStore::IGetter& iGetter) {
+  iBooker.setCurrentFolder(effFolderName_);
+
+  //Get the existing histos
+  MonitorElement* vtx_v_lumi = iGetter.get(vtxFolderName_ + "/NumberOfGoodPVtxVsLS_GenTk");
+
+  MonitorElement* scalLumi_v_lumi = iGetter.get(instLumiFolderName_ + "/lumiVsLS");
+
+  MonitorElement* eff_v_lumi_forward =
+      iGetter.get(effFolderName_ + "/hitefficiency_per_Lumisection_per_PXDisk_PXForward");
+
+  MonitorElement* eff_v_lumi_barrel =
+      iGetter.get(effFolderName_ + "/hitefficiency_per_Lumisection_per_PXLayer_PXBarrel");
+
+  if (!vtx_v_lumi) {
+    edm::LogWarning("SiPixelPhase1EfficiencyExtras")
+        << "no NumberOfGoodPVtxVsLS_GenTK ME is available in " << vtxFolderName_ << std::endl;
+    return;
+  } else if (!scalLumi_v_lumi) {
+    edm::LogWarning("SiPixelPhase1EfficiencyExtras")
+        << "no lumiVsLS ME is available in " << instLumiFolderName_ << std::endl;
+    return;
+  }
+
+  //Get the max value of inst lumi for plot
+  int yMax2 = scalLumi_v_lumi->getTProfile()->GetMaximum();
+  yMax2 = yMax2 + yMax2 * .1;
+
+  //Book new histos
+  MonitorElement* eff_v_vtx_barrel =
+      iBooker.book2D("hitefficiency_per_meanNvtx_per_PXLayer_PXBarrel",
+                     "hitefficiency_per_meanNvtx_per_PXLayer_PXBarrel; meanNvtx; PXLayer",
+                     500,
+                     0,
+                     100,
+                     3,
+                     .5,
+                     3.5);
+
+  MonitorElement* eff_v_vtx_forward =
+      iBooker.book2D("hitefficiency_per_meanNvtx_per_PXDisk_PXForward",
+                     "hitefficiency_per_meanNvtx_per_PXDisk_PXForward; meanNvtx; PXDisk",
+                     500,
+                     0,
+                     100,
+                     7,
+                     -3.5,
+                     3.5);
+
+  MonitorElement* eff_v_scalLumi_barrel =
+      iBooker.book2D("hitefficiency_per_scalLumi_per_PXLayer_PXBarrel",
+                     "hitefficiency_per_scalLumi_per_PXLayer_PXBarrel; scal inst lumi E30; PXLayer",
+                     500,
+                     0,
+                     yMax2,
+                     3,
+                     .5,
+                     3.5);
+
+  MonitorElement* eff_v_scalLumi_forward =
+      iBooker.book2D("hitefficiency_per_scalLumi_per_PXDisk_PXForward",
+                     "hitefficiency_per_scalLumi_per_PXDisk_PXForward; scal inst lumi E30; PXDisk",
+                     500,
+                     0,
+                     yMax2,
+                     7,
+                     -3.5,
+                     3.5);
+
+  //initialize variables
+  int numLumiNvtx = int(vtx_v_lumi->getTH1()->GetNbinsX());
```

Comment:
can't the `getTH1()` here and elsewhere be avoided using directly the thread-safe `MonitorElement` interfaces?

Reply:
Fixed!